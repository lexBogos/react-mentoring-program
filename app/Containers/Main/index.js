import React from "react";
import FilterPanel from "../../Components/FilterPanel";
import MoviesList from "../../Components/MoviesList";
import ErrorBoundary from "../../Components/ErrorBoundary";
import "./index.scss";

const Main = ({ movieList, setModalActive, setModalFields, movieListHook, setChoosenMovie }) => (
  <div className="mainContainer">
    <FilterPanel movieList={movieList} movieListHook={movieListHook} />
    <ErrorBoundary>
      <MoviesList
        movieList={movieList}
        setModalActive={setModalActive}
        setModalFields={setModalFields}
        movieListHook={movieListHook}
        setChoosenMovie={setChoosenMovie}
      />
    </ErrorBoundary>
  </div>
);

export default Main;
