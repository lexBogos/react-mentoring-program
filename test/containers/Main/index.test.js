import React from 'react';
import Main from '../../../app/containers/Main/index';
import '@testing-library/jest-dom/extend-expect';
import { createStore } from "redux";
import reducer from "../../../app/reducers/index";
import { Provider } from "react-redux";

import { render as rtlRender } from '@testing-library/react';

function render(
    ui,
    {
      initialState,
      store = createStore(reducer, initialState),
      ...renderOptions
    } = {}
  ) {
    function Wrapper({ children }) {
      return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  }

describe('Main component', () => {
        it('render Main component toMatchSnapshot', () => {
            const { asFragment } = render((<Main/>));
            expect(asFragment(<Main/>)).toMatchSnapshot();
        })
})