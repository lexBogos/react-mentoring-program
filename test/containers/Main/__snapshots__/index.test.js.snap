// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component render Main component toMatchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <div
      class="filterPanel"
    >
      <div
        class="typeList"
      >
        <a
          class="filterLink"
          href="#"
        >
          ALL
        </a>
        <a
          class="filterLink"
          href="#"
        >
          DOCUMENTARY
        </a>
        <a
          class="filterLink"
          href="#"
        >
          COMEDY
        </a>
        <a
          class="filterLink"
          href="#"
        >
          HORROR
        </a>
        <a
          class="filterLink"
          href="#"
        >
          CRIME
        </a>
      </div>
      <div
        class="sortFilter"
      >
        <label
          class="sortFilterLabel"
          for="filterList"
        >
          SORT BY
        </label>
        <select
          class="filterList"
          id="filterId"
          name="filterList"
        >
          <option
            value="date"
          >
            RELEASE DATE
          </option>
          <option
            value="name"
          >
            NAME
          </option>
          <option
            value="rate"
          >
            RATE
          </option>
        </select>
      </div>
    </div>
    <h1>
      Smth went wrong!
    </h1>
  </div>
</DocumentFragment>
`;
